modules/openapi-generator/src/main/resources/go/model_simple.mustache

```diff
@@ -356,20 +377,42 @@ func (o *{{{classname}}}) UnmarshalJSON(data []byte) (err error) {
 {{/requiredVars}}
 	}
 
+	defaultValueFuncMap := map[string]func() interface{} {
+{{#requiredVars}}
+{{#useDefaultValuesForRequiredVars}}
```

Comment:
The existing generated code should not have any diff if the `useDefaultValueForRequiredVars` is false. From that perspective, should this be wrapping all the changed code?

Reply:
I have wrapped whole newly added code under useDefaultValueForRequiredVars. Could you please check.

Reply:
@wing328  @vvb  all test cases passed. Could you please review.